test/models/user_test.rb

```diff
@@ -475,4 +475,84 @@ class UserTest < ActiveSupport::TestCase
     assert_raise(ActiveRecord::RecordNotFound) { collection.reload }
     assert_raise(ActiveRecord::RecordNotFound) { node.reload }
   end
+
+  # to run this test run the following command
+  # docker compose run test bin/rails test test/models/user_test.rb --name '/changing_role_to_registered_user_role_with_approved_events_count_greater_than_threshold_is_invalid/'
+  # test to check if user has updated role and approved_event_count
+  test 'changing_role_to_registered_user_role_with_approved_events_count_greater_than_threshold_is_invalid' do
+    @registered_user_role = Role.find_by!(title: 'Registered user')
+    @user = users(:trusted_user)
+
+    assert_raises(ActiveRecord::RecordNotSaved) do
+      # Change role to Registered user and set approved_events_count > threshold
+      @user.update!(role: @registered_user_role, approved_events_count: 11)
+    end
+
+    assert_includes @user.errors.full_messages,
+                    "A 'registered_user' cannot have approved events count more than #{User::EVENT_APPROVAL_THRESHOLD}. Please change the input accordingly."
```

Comment:
it's a bit brittle to include the expected error message as the full text in the tests. here, we're testing whether the user gets the right information in the error message, so checking for the key words is enough (and more flexible). alternatively, we can check against a constant containing the message, testing the behavior of choosing the right error message and not the meaning of it.